sailfishusermanager.cpp

```diff
@@ -556,11 +555,13 @@ void SailfishUserManager::onUnitJobFailed(SystemdManager::Job &job, SystemdManag
         // autologind was started but starting user@.service failed, probably because it was already starting
         qCWarning(lcSUM) << "Starting session systemd failed, is it already starting?";
         m_switchUser = 0;
+    } else {
+        // Inform UI
+        emit currentUserChangeFailed(m_switchUser);
```

Comment:
Just wondering... do we get notifications only about jobs started from here? Or can some random unrelated job failure lead into this 'else' block and trigger user switch failure actions?

Reply:
Been testing this a while and at least so far been unable to get errors.

Reply:
Those four cases above should handle all possible cases. This else block is never reached as only those jobs that are started by user-managerd are handled here.

Reply:
Dropped the else block. Of course now when testing I cannot get these to fail so this is more or less just guesswork. I think this signal is still about errors, now there's just someone listening too and showing it in UI. In my book it's an error if nothing happens when requesting user switch.

Reply:
> In my book it's an error if nothing happens when requesting user switch.
I agree. That's why there was the TODO comment that it should be signaled some way in that particular spot. Also in the bug report there is a suggestion of what kind of signal it could be. I just think that in these two cases we should make some difference there so UI can act the way it needs to. I mean the other failures are really fatal but this one is not so much.